core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2060,6 +2060,25 @@ private RexNode simplifyOrs(List<RexNode> terms, RexUnknownAs unknownAs) {
           }
         }
         break;
+      case IS_NOT_TRUE:
+        if (terms.contains(((RexCall) term).getOperands().get(0))) {
+          return trueLiteral;
```

Comment:
I'm not sure about this; I think safety should still be considered:
```
1/0 or x or (x is not true) 
```
I think even if its okay for `x` ; we can't replace it with `true` unless all other parts of the or passes `isSafeExpression` or something similar

Reply:
Same viewpoint as my other comment.

Reply:
My position on safety is that only CASE has a defined evaluation order. Though we've never really implemented it.

Reply:
Thanks for your explaination, I got your points. 
For the first case `x or not(x)`, I add a new limitation where `x` must be safeExpression.
But for the second case `a or x or not(x) or b`, I think we can simplify x or not(x) no matter whether `a` and `b` safe or not.
Thinking about a simple case: `true or unSafeUdf()`,  calcite will simplify this expression to `true`. And I have tested `select * from t where 1/0 or true` expression in mysql and sparksql, both of them will simpilfy this expression to `true`. And third, in most of programing language(like Java, C++, Python), they always simplify `true or someConditions` to `true` in if clause (A little different, it is Short-circuit OR actully, but we check all items).
So I think we don't need to care about outside items whether safe or not.